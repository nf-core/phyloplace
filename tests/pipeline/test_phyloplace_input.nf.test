nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_phyloplace_input"
    tag "pipeline"

    test("phyloplace input csv") {
        config "./nextflow.config"
        when {
            params {
                outdir           = "$outputDir"
                phyloplace_input = 'https://raw.githubusercontent.com/nf-core/test-datasets/phyloplace/testdata/phyloplace_input.csv'
                max_cpus         = 2
                max_memory       = '6.GB'
                max_time         = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/nf_core_phyloplace_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/epang/nuc_hmmer.epa_result.jplace.gz").exists() },
                { assert new File("$outputDir/epang/nuc_mafft.epa_result.jplace.gz").exists() },
                { assert new File("$outputDir/epang/PF14720mafft.epa_result.jplace.gz").exists() },
                { assert new File("$outputDir/epang/PF14720wo.epa_result.jplace.gz").exists() },
                { assert new File("$outputDir/epang/PF14720wt.epa_result.jplace.gz").exists() },
                { assert new File("$outputDir/gappa/nuc_hmmer.graft.nuc_hmmer.epa_result.newick").exists() },
                { assert new File("$outputDir/gappa/nuc_mafft.graft.nuc_mafft.epa_result.newick").exists() },
                { assert new File("$outputDir/gappa/PF14720mafft.graft.PF14720mafft.epa_result.newick").exists() },
                { assert new File("$outputDir/gappa/PF14720wo.graft.PF14720wo.epa_result.newick").exists() },
                { assert new File("$outputDir/gappa/PF14720wt.graft.PF14720wt.epa_result.newick").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
            )
        }
    }
}
