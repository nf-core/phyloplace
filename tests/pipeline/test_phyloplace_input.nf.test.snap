{
    "PF14720wt.graft.PF14720wt.epa_result.newick": {
        "content": [
            "PF14720wt.graft.PF14720wt.epa_result.newick:md5,1b9415cffdc5b744e7339be64436932e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T15:20:58.003255648"
    },
    "nuc_mafft.graft.nuc_mafft.epa_result.newick": {
        "content": [
            "nuc_mafft.graft.nuc_mafft.epa_result.newick:md5,bc977f0d29d1ba9596554a72ba3ac150"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T15:20:57.975292641"
    },
    "nuc_hmmer.graft.nuc_hmmer.epa_result.newick": {
        "content": [
            "nuc_hmmer.graft.nuc_hmmer.epa_result.newick:md5,e9d83a3df2176c78018cf3fc3367e388"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T15:20:57.958287198"
    },
    "PF14720mafft.graft.PF14720mafft.epa_result.newick": {
        "content": [
            "PF14720mafft.graft.PF14720mafft.epa_result.newick:md5,42eef418b47da35dd648b1b137030664"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T15:20:57.983957339"
    },
    "PF14720wo.graft.PF14720wo.epa_result.newick": {
        "content": [
            "PF14720wo.graft.PF14720wo.epa_result.newick:md5,1b9415cffdc5b744e7339be64436932e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T15:20:57.99303004"
    }
}